lib/modules/datasource/rubygems/common.ts

```diff
@@ -0,0 +1,43 @@
+import type { Http, SafeJsonError } from '../../../util/http';
+import type { AsyncResult } from '../../../util/result';
+import { joinUrlParts } from '../../../util/url';
+import type { ReleaseResult } from '../types';
+import { GemMetadata, GemVersions } from './schema';
+
+export function getV1Releases(
+  http: Http,
+  registryUrl: string,
+  packageName: string
+): AsyncResult<ReleaseResult, SafeJsonError> {
+  return http
+    .getJsonSafe(
+      joinUrlParts(registryUrl, '/api/v1/versions', `${packageName}.json`),
+      GemVersions
+    )
+    .transform(
+      (releases): Promise<ReleaseResult> =>
+        http
+          .getJsonSafe(
+            joinUrlParts(registryUrl, '/api/v1/gems', `${packageName}.json`),
+            GemMetadata
+          )
+          .transform(({ changelogUrl, sourceUrl, homepage }) => {
```

Comment:
looks okay, but I don't like the deep nesting.
can we extract some smaller functions to reduce the nesting?